mtags/src/main/scala-2/scala/meta/internal/pc/ConvertToNamedArgumentsProvider.scala

```diff
@@ -0,0 +1,33 @@
+package scala.meta.internal.pc
+
+import scala.meta.pc.OffsetParams
+
+import org.eclipse.{lsp4j => l}
+
+final class ConvertToNamedArgumentsProvider(
+    val compiler: MetalsGlobal,
+    params: OffsetParams,
+    numUnnamedArgs: Int
+) {
+  import compiler._
+  def convertToNamedArguments: List[l.TextEdit] = {
+    val unit = addCompilationUnit(
+      code = params.text(),
+      filename = params.uri().toString(),
+      cursor = None
+    )
+
+    val typedTree = typedTreeAt(unit.position(params.offset))
```

Comment:
> Maybe we'd want to consider the next enclosing apply (ie, addCompilationUnit) as a candidate?

You can access `lastVisitedParentTrees` to see the enclosing trees (make sure you access `lastVisitedParentTrees` after calling `typedTreeAt(pos)` or `locateTree(pos)` (as long as I remember)).
https://github.com/scalameta/metals/blob/d0cf15ced536460bca4d8df5e16042a8595fd4d5/mtags/src/main/scala-2/scala/meta/internal/pc/InferredTypeProvider.scala#L92-L97

Reply:
Now, `ConvertToNamedArgumentsProvider` receive the exact position of the target Apply, and no need to traverse trees